### DISCUSSION TITLE
anchors element

### DISCUSSION THREAD

**Discussion by wadsworj** [MyreMylar/pygame_gui#275] [Ideas]
Two questions/ideas about the anchors property that gets passed to most of these elements

1. is there any reason why all anchor elements need to be defined? ie, why can't I only define the 'left' anchor but use all other defaults?
2. are there any 'default' anchor layouts? (ie right align, center align, and left align). Most programs offer these three options at least:

example in word pad:
![image](https://user-images.githubusercontent.com/5513838/173072694-57494b57-686c-4e3c-b358-f2a3ef5ce6c7.png)

example in visual studio designer properties:
![image](https://user-images.githubusercontent.com/5513838/173073067-87319984-85b7-4848-9a24-664d7f8d4a48.png)

**Comment by wadsworj** [comment]
also wondering - is there any text alignment option?
I think I confused anchors with text alignment here.

**Comment by MyreMylar** [comment]
Hello,

There is text alignment for the UILabel, UITextBox or the UIButton (and thus also any text on buttons) in the theme file:

`text_horiz_alignment` for horizontal alignment and `text_vert_alignment` for vertical. this is relative to the rectangle you supply when creating the UILabel. You can read more about this [here](https://pygame-gui.readthedocs.io/en/latest/theme_reference/theme_label.html).

I expect the text box could be improved to mix different types of alignment within the same box (via something like `<P align="right">` to continue with the retro-style html parser I have here) at some point as the alignment is being done on a per line level behind the scenes.

**Reply by wadsworj** [reply]
Thanks, I'll need to understand the theming better.

For the properties that are exposed, I'm having trouble getting the alignment to update. What do I need to call to update the alignment when it is being set via a property?

```
test_text_entry.text_horiz_alignment = "left"
test_text_entry.text_vert_alignment = "left"
test_text_entry.rebuild_from_changed_theme_data()
```

I've tried the above but I'm not seeing the update at all:

![image](https://user-images.githubusercontent.com/5513838/173408106-b92632d9-542d-413a-9e76-692eeba4fd3c.png)

**Comment by MyreMylar** [comment]
You just want .rebuild() instead of rebuild from theme data.
